### PR TITLE
Models for logging participants and conversions 

### PR DESCRIPTION
**Pull request by kaedroho**
This PR adds:
 - Models for logging participants and conversions
 - Logic for deciding which variant to display to a new participant
 - Logic for serving the "treatment" variant instead of the "control" to the user (note, the "treatment" is the latest revision and the "control" is the live version of the page)
 - Logic for excluding bots and users with the ``DNT`` HTTP header from testing

### CODE REVIEW

File: wagtail_ab_testing/models.py [hunk 1]
```diff
@@ -8,20 +12,157 @@ def get_current_for_page(self, page):
...
+
+        self.save(update_fields=['status'])
+
+    def add_participant(self, variant=None):
+        """
+        Inserts a new participant into the log. Returns the variant that they should be shown.
+        """
+        # Get current numbers of participants for each variant
+        stats = self.hourly_logs.aggregate(
+            control_participants=Sum('participants', filter=Q(variant=self.Variant.CONTROL)),
```
Comment (KalobTaulien) [inline wagtail_ab_testing/models.py:58]: If we're doing filtering here, we're adding a WHERE clause the the SQL statement I believe. If this is the case, should we consider adding `db_index` to the relevant fields? Or would this likely never get big enough to warrant indexing the columns?

File: wagtail_ab_testing/models.py [hunk 2]
```diff
@@ -8,20 +12,157 @@ def get_current_for_page(self, page):
...
+        """
+        AbTestHourlyLog._increment_stats(self, variant, 0, 1)
+
+
+class AbTestHourlyLog(models.Model):
+    ab_test = models.ForeignKey(AbTest, on_delete=models.CASCADE, related_name='hourly_logs')
+    variant = models.CharField(max_length=9, choices=AbTest.Variant.choices)
+    date = models.DateField()
+    # UTC hour. Values range from 0 to 23
+    hour = models.PositiveSmallIntegerField()
```
Comment (KalobTaulien) [inline wagtail_ab_testing/models.py:106]: Points for using the smallest int field available with `PositiveSmallIntegerField`

**Review by KalobTaulien — APPROVED** [review]
Looking good. A couple of comments as food for thought. Otherwise this looks like its coming together nicely, awesome work! :tada:

File: wagtail_ab_testing/models.py (hunk 1 above)
Comment (kaedroho) [reply wagtail_ab_testing/models.py:58]: Good point, thanks! I'll add this in a subsequent PR!

Comment (kaedroho) [reply wagtail_ab_testing/models.py:58]: The existing ``unique_together`` on ``('ab_test', 'variant', 'date', 'hour'),`` should cover this. It's filtering by ``ab_test`` and ``variant`` which is a prefix of this index so it should get used for this query.

**Pull request merged**